### PR TITLE
Split into COG and NetCDF sub-packages

### PR DESCRIPTION
**Pull request by pjhartzell**
**Related Issue(s):**
#2 

**Description:**
This is a major reorganization that splits the package into two sub-packages: 

1. `netcdf`: Collection and Items describing the source NetCDF data files.
2. `cog`: Creates tiled COGs from the source NetCDF data files and Items describing each tile of COG data. The Items can optionally contain a `derived_from` link that points to a STAC Item describing the source NetCDF (e.g., an Item created by the `netcdf` sub-package). 

**PR checklist:**

- [x] Code is formatted (run `scripts/format`).
- [x] Code lints properly (run `scripts/lint`).
- [x] Tests pass (run `scripts/test`).
- [x] Documentation has been updated to reflect changes, if applicable.
- [x] Examples have been updated to reflect changes, if applicable
- [x] Changes are added to the [CHANGELOG](../CHANGELOG.md).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -11,77 +11,65 @@
...
   - [processing](https://github.com/stac-extensions/processing/)
-  - [proj](https://github.com/stac-extensions/projection/)
+  - [projection](https://github.com/stac-extensions/projection/)
   - [raster](https://github.com/stac-extensions/raster/)
-  - [version](https://github.com/stac-extensions/version/)
+  - [scientific](https://github.com/stac-extensions/scientific)
 - Extra fields:
-  - None
+  - `esa_cci_lc:version`: Land cover product version.
+  - `esa_cci_lc:tile`: Geographic coordinate of the lower left corner of the COG tile, rounded to the nearest degree, e.g., N45W180.
```
Comment (gadomski) [inline README.md]: Is the `grid` extension usable here?

File: environment.yml [hunk 2]
```diff
@@ -6,4 +6,4 @@ dependencies:
   - conda-forge::gdal>=3.4
   - conda-forge::geos>=3.3
   - conda-forge::rasterio>=1.3
-  - conda-forge::libstdcxx-ng # gdal dependency. Make sure it's from the same channel as gdal
+  # - conda-forge::libstdcxx-ng # gdal dependency. Make sure it's from the same channel as gdal
```
Comment (gadomski) [inline environment.yml]:
```suggestion
```

Just delete if it's not needed, IMO.

File: setup.cfg [hunk 3]
```diff
@@ -32,10 +32,6 @@ install_requires =
     python-dateutil >= 2.7.0
     types-python-dateutil >= 2.7.0
     netCDF4 >= 1.6.0
```
Comment (gadomski) [inline setup.cfg]: IMO `pystac`, at least, should be an explicit dependency, because we use it directly in code.

**Review by gadomski — CHANGES_REQUESTED** [review]

File: setup.cfg (hunk 3 above)
Comment (pjhartzell) [reply setup.cfg]: done.

File: README.md (hunk 1 above)
Comment (pjhartzell) [reply README.md]: Seems to fit. See https://github.com/stactools-packages/esa-cci-lc/blob/c64ee38c4325ec9db99c2824054955f020b7cb8c/examples/esa-cci-lc/C3S-LC-L4-LCCS-Map-300m-P1Y-2018-v2.1.1-N79W180/C3S-LC-L4-LCCS-Map-300m-P1Y-2018-v2.1.1-N79W180.json#L9

**Review by gadomski — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (pjhartzell) [reply README.md]: Based on our offline discussion today, I'm removing the grid extension. We are using a custom tiling scheme with a user defined tile size, so not a compelling use case. Incidentally, I am kicking the tires on the the grid extension on LCMAP, which follows the Landsat ARD tile grid.

**Pull request merged**